src/plugins/condition/components/ConditionEdit.vue

```diff
@@ -239,17 +275,46 @@ export default {
                 this.openmct.objects.get(this.condition.definition.criteria[0].key).then((obj) => {
                     this.telemetryObject = obj;
                     this.telemetryMetadata = this.openmct.telemetry.getMetadata(this.telemetryObject).values();
-                    this.selectedMetaDataKey = '';
-                    this.selectedTelemetryKey = this.telemetryObject.identifier;
+                    this.selectedMetaDataKey = this.getTelemetryMetadataKey();
+                    this.selectedTelemetryKey = this.getTelemetryKey();
                 });
             } else {
                 this.telemetryObject = null;
             }
         },
+        getTelemetryMetadataKey() {
+            let index = -1;
+            if (this.condition.definition.criteria[0].metaDataKey) {
+                index = _.findIndex(this.telemetryMetadata, (metadata) => {
+                    return metadata.key === this.condition.definition.criteria[0].metaDataKey;
+                });
+            }
+            return this.telemetryMetadata.length && index > -1 ? this.telemetryMetadata[index].key : '';
+        },
+        getTelemetryKey() {
+            let index = -1;
+            if (this.condition.definition.criteria[0].key) {
+                index = _.findIndex(this.telemetry, (obj) => {
+                    let key = this.openmct.objects.makeKeyString(obj.identifier);
+                    let conditionKey = this.openmct.objects.makeKeyString(this.condition.definition.criteria[0].key);
+                    return key === conditionKey;
+                });
+            }
+            return this.telemetry.length && index > -1 ? this.telemetry[index].identifier : '';
+        },
         hasTelemetry() {
             return this.condition.definition.criteria.length && this.condition.definition.criteria[0].key;
         },
+        updateConditionCriteria() {
+            if (this.condition.definition.criteria.length) {
+                this.condition.definition.criteria[0].key = this.selectedTelemetryKey;
+                this.condition.definition.criteria[0].metaDataKey = this.selectedMetaDataKey;
+                this.condition.definition.criteria[0].operation = this.selectedOperationKey;
+                this.condition.definition.criteria[0].input = [this.operationValue];
```

Comment:
Generally it's a good practice to define a variable for an object once if it's going to be accessed more than once. 
`let criteria = this.condition.definition.criteria[0];`

Reply:
Yes agreed. This was going to change soon with the addCriteria work that Joel is working on so I left it as is. I'll change it.

Reply:
Done